qutip/solver/heom/bofin_baths.py

```diff
@@ -337,6 +349,64 @@ def combine(cls, exponents, rtol=1e-5, atol=1e-7):
 
         return new_exponents
 
+    def spectral_density_approx(self, w, beta):
+        J = np.real(
+            self.power_spectrum(w) /
+            (((1 / (np.e**(w * beta) - 1)) + 1) * 2)
+        )
+        return J
+
+    def power_spectrum(self, w, beta=None):
+        """Calculates the power spectrum from the spectral density
+        or exponents if the spectral density is not available"""
+        try:
+            S = self.spectral_density(
+                w)*(((1 / (np.e**(w * beta) - 1)) + 1) * 2)
+        except (NameError, AttributeError):
+            # the next couple lines fail in some cases
+            # like underdamped spectral densities, and
+            # the expression that generates the power
+            # spectrum from components needs to change
+            S = np.zeros(len(w), dtype=np.complex128)
+            def s(x, y): return 2 * x * np.real(y) / (
+                (w - np.imag(y)) ** 2 + (np.real(y)) ** 2)
+            for exp in self.exponents:
+                if (
+                    exp.type == BathExponent.types['R'] or
+                    exp.type == BathExponent.types['RI']
+                ):
+                    S += s(exp.ck, exp.vk)
+                elif exp.type == BathExponent.types['RI']:
+                    S += s(1j*exp.ck2, exp.vk)
+                else:
+                    S += s(1j*exp.ck, exp.vk)
+        return S
+
+    def correlation_function(self, t, full=1):
+        """Computes the correlation function
```

Comment:
These and the other public facing docstrings should document the meanings and types of the arguments and the return value.

Reply:
Got it, need to improve all of those docstrings :)